### PR TITLE
feat(EPMGCIPUVA-3): Admin manages UVA users

### PR DESCRIPTION
**Pull request by andrei-roh**
image
image
image

### CODE REVIEW

File: packages/frontend/src/translations/Russian.json
Comment (hhharmEpm) [inline packages/frontend/src/translations/Russian.json:1]: I'm not sure, perhaps you were planning to make it in separate commit?  "usersList": "Пользователи" should be changed "usersList": "Список пользователей" according to story (line 19)

File: packages/frontend/src/utils/getSearch.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { IUser } from 'src/types';
+
+export const getSearch = (users: IUser[], value?: string) => {
```
Comment (hhharmEpm) [inline packages/frontend/src/utils/getSearch.ts:3]: I would change it to request to Firestore instead of making filtering on frontend side, because we make app with the idea of scalability in mind since UVA team said they want to add more countries in the future, so there'll be more people. And while searching in frontend works faster it may start causing problems where there'll be 100+ people.

I've found some example of how such search can be done here:

https://stackoverflow.com/questions/46568142/google-firestore-query-on-substring-of-a-property-value-text-search
```javascript
collectionRef
    .where('name', '>=', value)
    .where('name', '<=', value + '\uf8ff')
```

And also some sample from chatGPT, not sure it really works, but can be used as some example:
```javascript
import { collection, query, where, getDocs, orderBy, startAt, endAt } from 'firebase/firestore';

// Предполагается, что db - это ссылка на ваш Firestore экземпляр
const db = firebase.firestore();

const getUsersByName = async (name) => {
    // Коллекция, в которой хранятся пользователи
    const usersRef = collection(db, 'users');
    
    // Создаем запрос для поиска пользователей, имя которых начинается с введенных символов
    // Для этого используем orderBy и startAt для указания начального значения
    // и endAt для указания конечного значения, добавляя '\uf8ff' для включения всех символов после введенных
    const queryStartsWith = query(usersRef, orderBy('name'), startAt(name), endAt(name + '\uf8ff'));
    
    // Выполняем запрос
    const querySnapshot = await getDocs(queryStartsWith);
    
    // Массив для хранения результатов
    const usersList = querySnapshot.map((doc) => doc.data());
    
    return usersList;
};
```

File: packages/frontend/src/translations/Russian.json
Comment (andrei-roh) [reply packages/frontend/src/translations/Russian.json:1]: We discussed about it with Julia Kasyanava - now we are using "Пользователи"

File: packages/frontend/src/utils/getSearch.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  if (users.length === 0 || !value) return users;
+
+  return users.filter(
```
Comment (hhharmEpm) [inline packages/frontend/src/utils/getSearch.ts:8]: Let's cast value to lowerCase too to make search case-insensitive?

File: packages/frontend/src/utils/getSearch.ts (hunk 1 above)
Comment (hhharmEpm) [reply packages/frontend/src/utils/getSearch.ts:3]: or let's go with this variant unless Renat says otherwise)

File: packages/frontend/src/utils/getSearch.ts (hunk 2 above)
Comment (andrei-roh) [reply packages/frontend/src/utils/getSearch.ts:8]: Done

**Review by hhharmEpm — APPROVED** [review]

**Review by epmtesla — APPROVED** [review]
